for_review/logger.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import shutil
+from logging.handlers import RotatingFileHandler
+
+from colorlog import ColoredFormatter
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class DuplicateFilter(logging.Filter):
+    def filter(self, record):
+        repeated_number_exists = getattr(self, "repeated_number", None)
+        current_log = (record.module, record.levelno, record.msg)
+        if current_log != getattr(self, "last_log", None):
+            self.last_log = current_log
+            if repeated_number_exists:
+                LOGGER.warning(f"Last log repeated {self.repeated_number} times.")
+
+            self.repeated_number = 0
+            return True
+        if repeated_number_exists:
+            self.repeated_number += 1
+        else:
+            self.repeated_number = 1
+        return False
+
+
+def separator(symbol_, val=None):
+    terminal_width = shutil.get_terminal_size(fallback=(120, 40))[0]
+    if not val:
+        return f"{symbol_ * terminal_width}"
+
+    sepa = int((terminal_width - len(val) - 2) // 2)
+    return f"{symbol_ * sepa} {val} {symbol_ * sepa}"
+
+
+def setup_logging(log_level, log_file="/tmp/pytest-tests.log"):
```

Comment:
we should get name here and use it in line 39
`logger_obj = logging.getLogger(name)`

Reply:
Done! Please review.